autoPager.js

```diff
@@ -0,0 +1,112 @@
+const { env } = require('./config.js');
+const { compile } = require('handlebars');
+const config = require('./config.js')[env];
+const bdk = require('@salesforce/refocus-bdk')(config);
+const pagerDuty = require('./pagerDuty.js');
+
+/**
+ * @param {string} name - name of function where error occurred.
+ * @param {any} returnValue - value to be returned after logging.
+ * @returns {any} - whatever was passed as the returnsValue param.
+ */
+function logInvalidargs(name, returnValue) {
+  bdk.log.error(`Invalid args passed to function ${name}`);
+  return returnValue;
+}
+
+/**
+ * @param {number} roomId
+ * @param {string} botId
+ * @returns {Array<Incident>}
+ */
+async function getIncidentsForRoom(roomId, botId) {
+  const incidents = await bdk.getBotData(roomId, botId, 'onCallIncidents');
+  if (!incidents?.body?.[0].value) return [];
+  try {
+    const parsedIncidents = JSON.parse(incidents.body[0].value)?.incidents;
+    return parsedIncidents;
+  } catch (e) {
+    bdk.log.error(`Room ${roomId} - Failed to parse incidents when autopaging.`, e);
+    return undefined;
+  }
+}
+
+/**
+ * @param {string} roomId
+ * @returns {Array<object>} - teams to autoPage from room settings
+ */
+async function getTeamsToAutoPage(roomId) {
+  const room = await bdk.getRoomById(roomId);
+  const teams = room?.settings?.autoPageTeams;
+  return teams;
+}
+
+/**
+ * @param {Array<Incident>} incidentList - list of pager incidents that have already occurred.
+ * @param {Array<Team>} teamsToPage - list of teams to autopage, from room settings.
+ * @returns {Array<Team>} - list of teams from teamsToPage which have not been already paged.
+ */
+function removeAlreadyPagedTeams(incidentList, teamsToPage) {
+  if (!incidentList || !teamsToPage?.length)
+    return logInvalidargs('removeAlreadyPagedTeams', []);
+  const pagedTeams = incidentList.map((incident) => incident.service.id);
+  return teamsToPage.filter((team) => !pagedTeams.includes(team.id));
+}
+
+/**
+ * @param {Array<Team>} teams - list of teams to autopage, from room settings.
+ * @param {string} severity - current severity of room.
+ * @returns {Array<Team>} - array of teams which have the matching severity.
+ */
+function removeTeamsWithoutMatchingSeverity(teams, severity) {
+  if (!teams)
+    return logInvalidargs('removeTeamsWithoutMatchingSeverity', []);
+  teams.forEach((team) => {
+    team.severities = team.severities.map((sev) => sev.toLowerCase());
```

Comment:
just doing this for filter, otherwise some severities could be capitalised and missed

Reply:
Wouldnt work if doing this into the filter instead of mapping?

`teams.filter((team) => team.severities.toLowerCase().includes(severity.toLowerCase()))`

Reply:
I don't think you can call toLowerCase on an Array, which team.severities is

Reply:
ah, ok severities is [] - agreed!